app/controllers/sessions_controller.rb

```diff
@@ -3,7 +3,8 @@
 class SessionsController < ApplicationController
   layout 'auth', only: %i[new create]
 
-  before_action :redirect_to_home, if: :logged_in?, only: %i[new create]
+  before_action :require_no_authentication, only: %i[new create]
+  before_action :require_autentication, only: :destroy
```

Comment:
this implementation much better than the previous one ;)  but there is a small thing, we prefer the `ensure....` term than the `require....`, so if you could change to 

```
ensure_no_authentication
ensure_authentication
```

that will be great ❤️

Reply:
Fixed in 1e7d2746a5dd79bca9d2604188c7ba596d8fd1e2